homeassistant/components/yeelight/light.py

```diff
@@ -543,6 +561,7 @@ def set_effect(self, effect) -> None:
 
```

Comment:
We should add an `else: effect = None` here so we don't set `self._effect` unless we set a flow. 

Also you're changing anything when effect is `EFFECT_STOP`

This whole method, can you change the initial check from `if effect` to `if not effect: return` to save on indentation.

Reply:
Added early return on top. 

I think now we don't have to add else effect, as it wont reach that code, because of early return on top. 

I'm not sure what do you mean about EFFECT_STOP, https://github.com/home-assistant/home-assistant/pull/28975/files#diff-6b4ac627753ee21cde162f000848ba3bR532 it returns here, and if bulb stops flow, current effect will be set to None, in update method.